sources/web/datalab/static/idle-timeout.js

```diff
@@ -0,0 +1,264 @@
+      // UI, so we show nothing at all.
+      _debugLog('== no idleTimeoutSeconds');
+      dropdown.find('#idleTimeoutEnabledButton').hide();
+      dropdown.find('#idleTimeoutDisabledButton').hide();
+    } else {
+      _debugLog('== show disabled button');
+      dropdown.find('#idleTimeoutEnabledButton').hide();
+      dropdown.find('#idleTimeoutDisabledButton').show();
+    }
+  }
+
+  // Sets a timer to run updateTimeoutInfo.
+  function _setUpdateTimeoutInfoTimeout(dropdown) {
+    const interval = (dropdown && dropdown.hasClass('open')) ?
+        updateTimeoutInfoInterval : updateTimeoutInfoNoDisplayInterval;
+    if (timeoutInfoTimeout) {
+      window.clearTimeout(timeoutInfoTimeout);
+    }
+    timeoutInfoTimeout = window.setTimeout(() => {
+        updateTimeoutInfo(dropdown);
+    }, interval);
+  }
+
+  // Queries the server for the timeout info and then updates the display,
+  // or just updates the display if not enough time has passed to call the
+  // service again.
+  function updateTimeoutInfo(dropdown) {
+    const now = Date.now();
+    if (now - lastUpdateTimeoutTime > queryTimeoutInfoInterval) {
+      _debugLog('== Querying timeout');
+      const timeoutInfoUrl = window.location.protocol + "//" + window.location.host + "/_timeout";
+      function callback() {
+        _debugLog('== _timeout call response:');
+        _debugLog(this);
+        lastUpdateTimeoutTime = Date.now();
+        const result = this.response; // 'this' is the XHR
+        timeoutInfo = JSON.parse(result) || {};
+        timeoutInfo.expirationTime = Date.now() + timeoutInfo.secondsRemaining * 1000;
+        _updateTimeoutDisplay(dropdown);
+        _setUpdateTimeoutInfoTimeout(dropdown);
+      }
+      function errorHandler() {
+        _debugLog('== xhr error response:');
+        _debugLog(this);
+        const status = this.status;   // 'this' is the XHR
+        _debugLog('== status=' + status);
+        if (status == 0) {    // We get 0 when we lost our connection to the server.
+          _alertIfTimedOut();
+        }
+      }
+      const xhrOptions = {
+        errorCallback: errorHandler
+      };
+      xhr(timeoutInfoUrl, callback, xhrOptions);
+    } else {
+      // Too soon to query, just run the clock locally.
+      _updateTimeoutDisplay(dropdown);
+      _setUpdateTimeoutInfoTimeout(dropdown);
+    }
+  }
+
+  // Sends a message to the server to enable or disable the timout.
+  function toggleIdleTimeout() {
+    const newValue = timeoutInfo.enabled ? "false" : "true";
+    const timeoutUrl = window.location.protocol + "//" + window.location.host + "/_timeout?enabled=" + newValue;
+    const dropdown = $(this).parent().parent().parent();  // Walk up to the account drop-down.
+    _debugLog('== Changing enabled to ' + newValue);
+    updateTimeoutInfo(dropdown);
+    xhr(timeoutUrl, () => {
+      timeoutInfo.enabled = newValue;   // Display the new value right away.
```

Comment:
It is there a couple of lines below. Or did you mean something besides a call to updateTimeoutInfo?